test/unit-tests/GovernanceTimelock.t.sol

```diff
@@ -0,0 +1,268 @@
+    function test_execute_transaction_expired() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 0, bytes("42"));
+        skip(timelock.delay() + timelock.GRACE_PERIOD() + 1);
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.TX_EXPIRED.selector);
+        timelock.executeTransaction(1, address(42), 0, bytes("42"), eta);
+    }
+
+    /// @dev cannot execute tx with invalid value
+    function test_execute_transaction_invalid_value() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 1, bytes("42"));
+        skip(timelock.delay());
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.INVALID_MSG_VALUE.selector);
+        timelock.executeTransaction(1, address(42), 1, bytes("42"), eta);
+    }
+
+    /// @dev failed to execute tx on dst chain
+    function test_execute_transaction_fails_on_dst() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        // Use admin as dummy target address
+        timelock.scheduleTransaction(admin, 0, bytes("42"));
+        skip(timelock.delay());
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.EXECUTION_FAILS_ON_DST.selector);
+        timelock.executeTransaction(1, admin, 0, bytes("42"), eta);
+    }
+
+    /// @dev sets delay
+    function test_set_delay() public {
+        vm.startPrank(address(timelock));
+
+        uint256 oldDelay = timelock.delay();
+        vm.expectEmit(true, true, true, true, address(timelock));
+        emit DelayUpdated(oldDelay, 7 days);
+
+        timelock.setDelay(7 days);
+    }
+
+    /// @dev only timelock can set delay
+    function test_set_delay_only_self() public {
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.INVALID_SELF_CALLER.selector);
+        timelock.setDelay(7 days);
+    }
+
+    /// @dev cannot set delay below minimum
+    function test_set_delay_below_minimum() public {
+        vm.startPrank(address(timelock));
+
+        uint256 minDelay = timelock.MINIMUM_DELAY();
+        vm.expectRevert(Error.INVALID_DELAY_MIN.selector);
+        timelock.setDelay(minDelay - 1);
+    }
+
+    /// @dev cannot set delay above maximum
+    function test_set_delay_above_maximum() public {
+        vm.startPrank(address(timelock));
+
+        uint256 maxDelay = timelock.MAXIMUM_DELAY();
+        vm.expectRevert(Error.INVALID_DELAY_MAX.selector);
+        timelock.setDelay(maxDelay + 1);
+    }
+
+    /// @dev sets admin
+    function test_set_admin() public {
+        vm.startPrank(address(timelock));
+
+        address oldAdmin = timelock.admin();
+        vm.expectEmit(true, true, true, true, address(timelock));
+        emit AdminUpdated(oldAdmin, address(42));
+
+        timelock.setAdmin(address(42));
+    }
+
+    /// @dev only timelock can set admin
```

Comment:
can we also add a test for calling `setAdmin()` fn through the scheduled transaction flow

Reply:
Done.